torchao/dtypes/aqt.py

```diff
@@ -564,43 +634,72 @@ def _quantized_linear_op(input_tensor, weight_qtensor, bias):
             weight_is_uint4 and
             weight_qtensor.dtype == torch.bfloat16 and
             len(weight_qtensor.shape) == 2 and
-            weight_qtensor.block_size[0] == 1 and
             weight_qtensor.zero_point_domain == ZeroPointDomain.FLOAT and
             weight_qtensor.layout == "tensor_core_tiled"
         ):
-            # groupwise int4 quantization
-            groupsize = weight_qtensor.block_size[-1]
+            if not _from_flinear:
+                weight_qtensor = weight_qtensor.t()
```

Comment:
n00b q: why does this require a transpose?

Reply:
this is to align the dimensions, for block_size so that we can get groupsize from block_size argument, see L662, and also related to L575. right now the `_quantized_linear` does not have a well-defined accepted weight dimensions